app/src/main/java/ch/epfl/sweng/eventmanager/ui/event/interaction/fragments/EventMainFragment.java

```diff
@@ -84,20 +88,27 @@ public void onResume() {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = super.onCreateView(inflater, container, savedInstanceState);
-
         if (view != null) ButterKnife.bind(this, view);
 
+        showcaseActivity = getParentActivity();
+
         // FIXME Handle NullPointerExceptions from the ChangeFragment
-        contactButton.setOnClickListener(v -> ((EventShowcaseActivity) getActivity()).changeFragment(new EventFormFragment(), true));
+        contactButton.setOnClickListener(v -> changeFromMainFragment(new EventFormFragment(), null));
 
-        news.setOnClickListener(v -> ((EventShowcaseActivity) getActivity()).changeFragment(new NewsFragment(), true));
+        news.setOnClickListener(v -> changeFromMainFragment(new NewsFragment(), showcaseActivity.newsFragment));
 
-        map.setOnClickListener(v -> ((EventShowcaseActivity) getActivity()).changeFragment(new EventMapFragment(),
-                true));
+        map.setOnClickListener(v -> changeFromMainFragment(new EventMapFragment(), showcaseActivity.eventMapFragment));
 
-        schedule.setOnClickListener(v -> ((EventShowcaseActivity) getActivity()).changeFragment(new ScheduleParentFragment(), true));
+        schedule.setOnClickListener(v -> changeFromMainFragment(new ScheduleParentFragment(), showcaseActivity.scheduleParentFragment));
 
         return view;
 
     }
+
```

Comment:
Better make a public changeFragment() method in the showcase activity as we don't have to have these weird public fragments

Reply:
Yes I got your point ok